test/e2e/certmanager/metrics_test.go

```diff
@@ -0,0 +1,291 @@
+				curlContainerName = createdPod.Spec.Containers[0].Name
+			}, util.Timeout, util.Interval).Should(gomega.Succeed())
+		})
+	})
+
+	ginkgo.AfterEach(func() {
+		gomega.Expect(util.DeleteNamespace(ctx, k8sClient, ns)).To(gomega.Succeed())
+		util.ExpectObjectToBeDeleted(ctx, k8sClient, resourceFlavor, true)
+		util.ExpectObjectToBeDeleted(ctx, k8sClient, metricsReaderClusterRoleBinding, true)
+		util.ExpectObjectToBeDeletedWithTimeout(ctx, k8sClient, curlPod, true, util.LongTimeout)
+		util.ExpectAllPodsInNamespaceDeleted(ctx, k8sClient, ns)
+	})
+
+	ginkgo.When("workload is admitted", func() {
+		var (
+			clusterQueue *v1beta1.ClusterQueue
+			localQueue   *v1beta1.LocalQueue
+			workload     *v1beta1.Workload
+		)
+
+		ginkgo.BeforeEach(func() {
+			clusterQueue = utiltesting.MakeClusterQueue("").
+				GeneratedName("test-cq-").
+				ResourceGroup(
+					*utiltesting.MakeFlavorQuotas(resourceFlavor.Name).
+						Resource(corev1.ResourceCPU, "1").
+						Resource(corev1.ResourceMemory, "1Gi").
+						Obj(),
+				).
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, clusterQueue)).To(gomega.Succeed())
+
+			localQueue = utiltesting.MakeLocalQueue("", ns.Name).
+				GeneratedName("test-lq-").
+				ClusterQueue(clusterQueue.Name).
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, localQueue)).To(gomega.Succeed())
+
+			workload = utiltesting.MakeWorkload("test-workload", ns.Name).
+				Queue(localQueue.Name).
+				PodSets(
+					*utiltesting.MakePodSet("ps1", 1).Obj(),
+				).
+				RequestAndLimit(corev1.ResourceCPU, "1").
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, workload)).To(gomega.Succeed())
+		})
+
+		ginkgo.AfterEach(func() {
+			util.ExpectObjectToBeDeleted(ctx, k8sClient, workload, true)
+			util.ExpectObjectToBeDeleted(ctx, k8sClient, clusterQueue, true)
+		})
+
+		ginkgo.It("should ensure the default metrics are available", func() {
+			util.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, workload)
+
+			metrics := [][]string{
+				{"kueue_admission_attempts_total"},
+				{"kueue_admission_attempt_duration_seconds"},
+				{"kueue_pending_workloads", clusterQueue.Name},
+				{"kueue_reserving_active_workloads", clusterQueue.Name},
+				{"kueue_admitted_active_workloads", clusterQueue.Name},
+				{"kueue_quota_reserved_workloads_total", clusterQueue.Name},
+				{"kueue_quota_reserved_wait_time_seconds", clusterQueue.Name},
+				{"kueue_admitted_workloads_total", clusterQueue.Name},
+				{"kueue_admission_wait_time_seconds", clusterQueue.Name},
+				{"kueue_cluster_queue_resource_usage", clusterQueue.Name},
+				{"kueue_cluster_queue_status", clusterQueue.Name},
+				{"kueue_cluster_queue_resource_reservation", clusterQueue.Name},
+				{"kueue_cluster_queue_nominal_quota", clusterQueue.Name},
+				{"kueue_cluster_queue_borrowing_limit", clusterQueue.Name},
+				{"kueue_cluster_queue_lending_limit", clusterQueue.Name},
+				{"kueue_cluster_queue_weighted_share", clusterQueue.Name},
+
+				// LocalQueueMetrics
```

Comment:
I don't really see any feature gates for this test. Are we enabling the alpha feature gate for LocalQueueMetrics?

Or you are using an alpha config for these jobs?

Reply:
I actually took this test from `singlecluster/metrics_test.go`. That test uses `test/e2e/config/default`, and the cert-manager suite also inherits its default configuration from there, so I believe we're covered.

Reply:
I see. It works because of https://github.com/kubernetes-sigs/kueue/blob/main/test/e2e/config/default/manager_e2e_patch.yaml#L6.

Reply:
Yup

Reply:
I would suggest to simplify the test. The test you copied from aims to make sure the metrics code is working. The purpose of this test is just to make sure the metrics endpoints works. So, I think the simplest possible sanity test is enough. Eg, setup one LQ, one CQ, one RF, and admit one workload, then check kueue_quota_reserved_workloads_total is present.